azext_iot/central/providers/device_provider.py

```diff
@@ -194,20 +201,92 @@ def get_device_registration_info(
 
         return info
 
-    def get_all_registration_info(self, central_dns_suffix="azureiotcentral.com"):
+    def filter_dps_info(self, dps_info, device_status):
+        error = {
+            "provisioned": "None",
+            "registered": "Device it not yet provisioned.",
+            "blocked": "Device is blocked by admin",
+            "unassociated": "Device does not have a valid template associated with it",
+        }
+
+        filtered_dps_info = {
+            "status": dps_info.get("status"),
+        }
+
+        filtered_dps_info.update({"error": error.get(device_status)})
+        return filtered_dps_info
+
+    def update_device_info_filter_status(self, device, value):
+        if value is None:
+            return self.update_device_status(device)
+        updated_device_data = {
+            "id": device["id"],
+            "displayName": device.get("displayName"),
+            "instanceOf": device.get("instanceOf"),
+            "simulated": device.get("simulated"),
+            "deviceStatus": value,
+        }
+        return updated_device_data
+
+    def update_device_status(self, device):
+        if device["approved"] is False:
+            updated_device = self.update_device_info_filter_status(
+                device, DeviceStatus.blocked.value
+            )
+        else:
+            if device.get("instanceOf") is None:
+                updated_device = self.update_device_info_filter_status(
+                    device, DeviceStatus.unassociated.value
+                )
+            else:
+                if device["provisioned"] is False:
+                    updated_device = self.update_device_info_filter_status(
+                        device, DeviceStatus.registered.value
+                    )
+                else:
+                    updated_device = self.update_device_info_filter_status(
+                        device, DeviceStatus.provisioned.value
+                    )
+        return updated_device
+
+    def update_all_device_status(self, devices):
```

Comment:
try to make this function name a bit more specific, since we might need to do different kinds of updates in the future